src/test/kotlin/study/DslTest.kt

```diff
@@ -0,0 +1,157 @@
+            name("Seokjin")
+            company("dh")
+        }
+
+        assertThat(person.name).isEqualTo("Seokjin")
+        assertThat(person.company).isEqualTo("dh")
+    }
+
+    /*
+    * introduce {
+    *     skills {
+    *         soft("A passion for problem solving")
+    *         soft("Good communication skills")
+    *         hard("Kotlin")
+    *     }
+    * }
+    */
+    @Test
+    fun `skills test`() {
+        val person: Person = introduce {
+            name("Seokjin")
+            company("dh")
+            skills {
+                soft("A passion for problem solving")
+                soft("Good communication skills")
+                hard("Kotlin")
+            }
+        }
+        val expectedSoft = listOf("A passion for problem solving", "Good communication skills")
+        val expectedHard = listOf("Kotlin")
+
+        assertThat(person.name).isEqualTo("Seokjin")
+        assertThat(person.company).isEqualTo("dh")
+        assertThat(person.skills?.soft).hasSameElementsAs(expectedSoft)
+        assertThat(person.skills?.hard).hasSameElementsAs(expectedHard)
+    }
+
+    /*
+    * introduce {
+    *     languages {
+    *         "Korean" level 3
+    *         "English" level 3
+    *     }
+    * }
+    * */
+    @Test
+    fun `languages test`() {
+        val person: Person = introduce {
+            name("Seokjin")
+            company("dh")
+            skills {
+                soft("A passion for problem solving")
+                soft("Good communication skills")
+                hard("Kotlin")
+            }
+            languages {
+                "Korean" level 3
+                "English" level 3
+            }
+        }
+        val expectedSoft = listOf("A passion for problem solving", "Good communication skills")
+        val expectedHard = listOf("Kotlin")
+
+        assertThat(person.name).isEqualTo("Seokjin")
+        assertThat(person.company).isEqualTo("dh")
+        assertThat(person.skills?.soft).hasSameElementsAs(expectedSoft)
+        assertThat(person.skills?.hard).hasSameElementsAs(expectedHard)
+        assertThat(person.languages!!["Korean"]).isEqualTo(3)
+        assertThat(person.languages["English"]).isEqualTo(3)
+    }
+
+    private fun introduce(block: PersonBuilder.() -> Unit): Person {
+//        return PersonBuilder().apply { block() }.build()
+        return PersonBuilder().apply(block).build()
+    }
+
+
+
+    class PersonBuilder {
+        private lateinit var name: String
+        private var company: String? = null
+        private var skills: Skills? = null
+        private val languages = mutableMapOf<String, Int>()
+
+        fun name(value: String) {
+            name = value
+        }
+
+        fun company(value: String) {
+            company = value
+        }
+
+        fun skills(block: SkillsBuilder.() -> Unit) {
+            skills = SkillsBuilder().apply(block).build()
+        }
+
+        fun languages(block: Map<String, Int>.() -> Unit) {
```

Comment:
How about extract `LanguagesBuilder` like `SkillsBuilder`?
What benefits do you think that might bring?